x-pack/test/security_solution_api_integration/test_suites/detections_response/utils/retry_if_conflicts.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsErrorHelpers } from '@kbn/core/server';
+import { ToolingLog } from '@kbn/tooling-log';
+
+type RetryableForConflicts<T> = () => Promise<T>;
+
+// Number of times to retry when conflicts occur
+export const RetryForConflictsAttempts = 2;
+
+/*
+ * Retry an operation if it runs into 409 Conflicts,
+ * up to a maximum number of attempts.
+ */
+export async function retryIfConflicts<T>(
+  logger: ToolingLog,
+  name: string,
+  operation: RetryableForConflicts<T>,
+  retries: number = RetryForConflictsAttempts
+): Promise<T> {
+  // run the operation, return if no errors or throw if not a conflict error
+  try {
+    return await operation();
+  } catch (err) {
+    if (!SavedObjectsErrorHelpers.isConflictError(err)) {
+      throw err;
+    }
```

Comment:
Why are we handling only SO conflict errors here? The 409 error we saw in the test was from ES directly. I think additionally we need to handle ES errors of `"type":"version_conflict_engine_exception"`. Maybe these details don't matter and we could simply check the status code.

Reply:
See comment in PR